include/zephyr/drivers/i2c.h

```diff
@@ -746,13 +743,11 @@ __syscall int i2c_get_config(const struct device *dev, uint32_t *dev_config);
 
 static inline int z_impl_i2c_get_config(const struct device *dev, uint32_t *dev_config)
 {
-	const struct i2c_driver_api *api = (const struct i2c_driver_api *)dev->api;
-
-	if (api->get_config == NULL) {
+	if (DEVICE_API_GET(i2c, dev)->get_config == NULL) {
 		return -ENOSYS;
 	}
 
-	return api->get_config(dev, dev_config);
+	return DEVICE_API_GET(i2c, dev)->get_config(dev, dev_config);
```

Comment:
wouldn't you be better off keeping the `api` local variable in this case? Up to the maintainers really but I'd rather do

```c
	const struct i2c_driver_api *api = DEVICE_API_GET(i2c, dev);
```

if there's extra conditoins like null checks and not just the api call.

Anyway, up to you folks.

Reply:
Reverted the header changes for now as discussed in discord.